text/0000-mergeable-rustdoc-cross-crate-info.md

```diff
@@ -0,0 +1,439 @@
+│       │   ├── index.html
+│       │   └── sidebar-items.js
+│       ├── index.html
+│       ├── s
+│       │   ├── all.html
+│       │   ├── index.html
+│       │   ├── sidebar-items.js
+│       │   └── struct.S.html
+│       ├── search.desc
+│       │   ├── i
+│       │   │   └── i-desc-0-.js
+│       │   ├── s
+│       │   │   └── s-desc-0-.js
+│       │   └── t
+│       │       └── t-desc-0-.js
+│       ├── search-index.js
+│       ├── settings.html
+│       ├── src
+│       │   ├── i
+│       │   │   └── lib.rs.html
+│       │   ├── s
+│       │   │   └── lib.rs.html
+│       │   └── t
+│       │       └── lib.rs.html
+│       ├── src-files.js
+│       ├── static.files
+│           │   ├── COPYRIGHT-23e9bde6c69aea69.txt
+│           │   ├── favicon-2c020d218678b618.svg
+│           │   └── <rest of the contents excluded>
+│       ├── t
+│       │   ├── all.html
+│       │   ├── index.html
+│       │   ├── sidebar-items.js
+│       │   └── trait.T.html
+│       └── trait.impl
+│           ├── core
+│           │   ├── marker
+│           │   │   ├── trait.Freeze.js
+│           │   │   ├── trait.Send.js
+│           │   │   ├── trait.Sync.js
+│           │   │   └── trait.Unpin.js
+│           │   └── panic
+│           │       └── unwind_safe
+│           │           ├── trait.RefUnwindSafe.js
+│           │           └── trait.UnwindSafe.js
+│           └── t
+│               └── trait.T.js
+├── s
+│   ├── src
+│   │   └── lib.rs
+│   └── target
+│       ├── doc
+│       │   ├── help.html
+│       │   ├── .lock
+│       │   ├── s
+│       │   │   ├── all.html
+│       │   │   ├── index.html
+│       │   │   ├── sidebar-items.js
+│       │   │   └── struct.S.html
+│       │   ├── search.desc
+│       │   │   └── s
+│       │   │       └── s-desc-0-.js
+│       │   ├── settings.html
+│       │   └── src
+│       │       └── s
+│       │           └── lib.rs.html
+│       ├── doc.parts
+│       │   └── s
+│       │       └── crate-info.json
+│       └── libs.rmeta
+└── t
+    ├── src
+    │   └── lib.rs
+    └── target
+        ├── doc
+        │   ├── help.html
+        │   ├── .lock
+        │   ├── search.desc
+        │   │   └── t
+        │   │       └── t-desc-0-.js
+        │   ├── settings.html
+        │   ├── src
+        │   │   └── t
+        │   │       └── lib.rs.html
+        │   └── t
+        │       ├── all.html
+        │       ├── index.html
+        │       ├── sidebar-items.js
+        │       └── trait.T.html
+        ├── doc.parts
+        │   └── t
+        │       └── crate-info.json
+        └── libt.rmeta
+</pre>
+
+</details>
+
+## Suggested workflows
+
+### No cross-crate information 
+Provide `--merge=none` to every invocation of rustdoc.
+
+### Cross-crate information, mutate shared directory
+
+Use `--merge=read-write`, and specify the same `--out-dir` to every invocation of rustdoc. `--merge=read-write` will be the default value if `--merge` is not provided. This is the workflow that Cargo uses.
```

Comment:
Another thought I had: why not provide `read` and `write` values and possibly allow them to be separated by a comma like `--merge=read,write` or `--merge=write,read`. Like that, only two values. :)

Reply:
Hum, since `read` by itself is useless, not sure if it's a good idea or not...

Reply:
I think the command line help text is easier to write for `--merge=read-write|write-only|none`, since we don't have to explain that there are useless states.

Reply:
Yes I don't really see the need for it.